src/promptflow/promptflow/_sdk/operations/_run_operations.py

```diff
@@ -111,11 +111,13 @@ def _print_run_summary(self, run: Run) -> None:
         )
 
     @monitor_operation(activity_name="pf.runs.stream", activity_type=ActivityType.PUBLICAPI)
-    def stream(self, name: Union[str, Run]) -> Run:
+    def stream(self, name: Union[str, Run], raise_on_error: bool = True) -> Run:
```

Comment:
What's the context that we need to change this behavior? And shall we change azure stream behavior to make alignment?

Reply:
context: in local, if a run is failed or canceled, the logs might be empty, then `pf.stream` will show nothing; therefore, we add `raise_on_error` to raise exception in such case.

and I think I have synced the changes to azure stream?